test/C/src/DeadlineUpdate.lf

```diff
@@ -0,0 +1,64 @@
+// Test whether the lf_update_deadline() function works.
+// This test covers two scenarios:
+// 1. Updating to an earlier deadline: Verify that a 100 ms declared deadline
+// can be shortened to 50 ms at runtime, and that a violation occurs
+// accordingly.
+// 2. Updating to a later deadline: Verify that a 50 ms declared deadline
+// can be relaxed to 100 ms at runtime, and violation is triggered at
+// the updated time 100 ms.
+// The test prints SUCCESS if both cases behave as expected, otherwise it
+// reports an error and exits.
+target C
+
+reactor A {
+  state i: int
+  logical action a
+
+  // Test Case 1: Verify that decreasing the deadline works.
+  // The reaction starts with a declared deadline of 100 ms, but we update it
+  // to 50 ms at runtime. A deadline violation should therefore be detected
+  // around physical time 50 ms . If no violation is detected
+  // after 50 ms, the test fails.
+  reaction(startup) -> a {=
+    self->i = 0;
+    lf_update_deadline(self, MSEC(50));
+    while(!lf_check_deadline(self, true)){
```

Comment:
I don't think this needs a while loop here, and can be simplified.
Also, how about you reduce the sleep to lf_sleep(MSEC(10)) and set the initial deadline to 500 msec, to make this test less flaky and take less time to execute?

Reply:
Fixed this.